packages/@aws-cdk/aws-synthetics/test/code.test.ts

```diff
@@ -0,0 +1,124 @@
+import '@aws-cdk/assert/jest';
+import * as path from 'path';
+import { App, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+let app: App;
+beforeEach(() => {
+  app = new App();
+  stack = new Stack(app, 'canaries');
+});
+
+describe('Code.fromInline', () => {
+  test('basic use case', () => {
+    // WHEN
+    new synthetics.Canary(stack, 'Canary', {
+      test: synthetics.Test.custom({
+        handler: 'index.handler',
```

Comment:
Are you unit testing `Code` or `Canary` here? I don't understand what the `Canary` boilerplate adds here, besides making those tests fail when `Canary` is broken (making the root cause harder to identify)

Reply:
I removed some of them, but it seems necessary to add in canaries for a few of the Asset tests.